### PR TITLE
Blog Detail Backend Development

### PR DESCRIPTION
**Pull request by russellseanrs**
**Issue Link**
Asana: https://app.asana.com/0/1202718495875213/1202725278795548/f

**Definition of Done:**

- Added some changes on blog details, number of votes
- Also added new changes in model
- Comments and votes were optional so I didn't put it yet

**Notes:**
1. Go to http://127.0.0.1:8000/api/posts/
2. then go to http://127.0.0.1:8000/api/posts/second-post-1/
3. http://127.0.0.1:8000/api/posts/< slug >/
4. if you want to create go to http://127.0.0.1:8000/admin/ (add your post)

**Screenshots/Recordings:**
291617008_3271110989824556_8274374143636119047_n

291761570_1417285425415690_6771512825795499869_n

292407127_433494868734488_1044593876951136958_n

### CODE REVIEW

File: backend/post/views.py [hunk 1]
```diff
@@ -21,3 +22,9 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(instance)
         return Response(serializer.data)
 
+    def delete(self, request, *args, **kwargs):
+        post = Post.objects.filter(pk=kwargs['pk'], poster=self.request.user)
+        if post.exists():
+            return self.destroy(request, *args, **kwargs)
+        else:
+            raise ValidationError("This is not your post to delete")
```
Comment (kentlouise) [inline backend/post/views.py]: Add a new line after the raise statement.

**Review by kentlouise — COMMENTED** [review]
Please update the description of Asana and PR details.

Update also the title of the PR, based on the asana task title.

You can refer to this docs. https://docs.google.com/document/d/1f8MQXMskwLXVUmmRxxyJEQguZGddM2DuE1nf4s6xWKg/edit

**Review by kentlouise — COMMENTED** [review]
I think we should add the `__pycache__` in gitignore.
pycache

**Comment by russellseanrs** [comment]
> I think we should add the `__pycache__` in gitignore. pycache

is it okay to do it in the next pr sir? 
and in every pr we have today it has __pycache__ sir

**Comment by kentlouise** [comment]
> > I think we should add the `__pycache__` in gitignore. pycache
> 
> is it okay to do it in the next pr sir? and in every pr we have today it has **pycache** sir

@russellseanrs 
I think we should update this one Sir. Since once this will be merge, it will be included in the develop branch.

**Review by kentlouise — COMMENTED** [review]
Please remove the __pycache__ in `backend/post/migrations/__pycache__`. Same procedure.

1. cd to the migrations folder.
2. Run: git rm -rf ./__pycache__
3. Run: git add -A
4. Push commit.

sample

File: backend/post/views.py [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 from rest_framework.response import Response
+from rest_framework.exceptions import ValidationError
 from .serializers import PostSerializer
 from .models import Post
 
 
 # Create your views here.
 class PostView(viewsets.ModelViewSet):
-    queryset = Post.objects.order_by('-published_at')
+    queryset = Post.objects
```
Comment (carl-mate-sun-asterisk) [inline backend/post/views.py]: I think this should be: 
queryset = Post.objects.all()

File: backend/post/views.py [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
     serializer_class = PostSerializer
     lookup_field = 'slug'
 
 class PostDetailView(viewsets.ModelViewSet):
-    queryset = Post.objects.order_by('-published_at')
+    queryset = Post.objects
```
Comment (carl-mate-sun-asterisk) [inline backend/post/views.py]: Same here:
queryset = Post.objects.all() 
Please correct me if I'm wrong hehe thank you!

Comment (russellseanrs) [reply backend/post/views.py]: yes, I forgot I already changed it. Thanks

Comment (carl-mate-sun-asterisk) [reply backend/post/views.py]: Thank you. Looks good to me.

**Pull request merged**